src/libraries/Common/src/Roslyn/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -35,7 +57,11 @@ internal static partial class SyntaxValueProviderExtensions
     /// <c>context.SyntaxProvider.CreateSyntaxProviderForAttribute(nameof(CLSCompliantAttribute), (node, c) => node is ClassDeclarationSyntax)</c>
     /// will find the <c>C</c> class.
     /// </summary>
-    public static IncrementalValuesProvider<SyntaxNode> ForAttributeWithSimpleName(
+    /// <remarks>
+    /// Note: a 'Values'-provider of arrays are returned.  Each array provides all the matching nodes from a single <see
+    /// cref="SyntaxTree"/>.
+    /// </remarks>
+    public static IncrementalValuesProvider<(SyntaxTree tree, ImmutableArray<SyntaxNode> matches)> ForAttributeWithSimpleName(
```

Comment:
instead of a flattened list of all nodes found across all files.  we instead get a match per-tree that has results, and that match includes all the matches *within* that file.